lib/ansible/modules/hostname.py

```diff
@@ -160,7 +160,8 @@ def __init__(self, module):
         if self.use is not None:
             strat = globals()['%sStrategy' % STRATS[self.use]]
             self.strategy = strat(module)
-        elif self.platform == 'Linux' and ServiceMgrFactCollector.is_systemd_managed(module):
+        elif platform.system() == 'Linux' and ServiceMgrFactCollector.is_systemd_managed(module):
```

Comment:
`self.platform` is meant to match `platform.system()` for comparisons made by `get_platform_subclass`

So no need to ensure that a defined strategy is selected and that it is `Linux`, just use `platform.system()` directly.